src/runtime_src/core/tools/common/ReportDebugIpStatus.cpp

```diff
@@ -379,127 +318,17 @@ DebugIpStatusCollector::readAIMCounter(debug_ip_data* dbgIpInfo)
   ++debugIpNum[AXI_MM_MONITOR];
   aimResults.NumSlots = (unsigned int)debugIpNum[AXI_MM_MONITOR];
 
-#ifndef _WIN32
-  // read counter values
-  xrt_core::system::monitor_access_type accessType = xrt_core::get_monitor_access_type();
-  if(xrt_core::system::monitor_access_type::ioctl == accessType) {
-    std::string aimName("aximm_mon_");
-    aimName = aimName + std::to_string(dbgIpInfo->m_base_address);
-
-    std::vector<char> nameSysfsPath;
-    nameSysfsPath.resize(512);
-    xclGetSysfsPath(handle, aimName.c_str(), "name", nameSysfsPath.data(), 512);
-    std::string namePath(nameSysfsPath.data());
-
-    std::size_t pos = namePath.find_last_of('/');
-    std::string path = namePath.substr(0, pos+1);
-    path += "counters";
-
-    std::ifstream ifs(path.c_str());
-    if(!ifs) {
-      return;
-    }
-
-    const size_t sz = 256;
-    char buffer[sz];
-    std::memset(buffer, 0, sz);
-    ifs.getline(buffer, sz);
-
-    std::vector<uint64_t> valBuf;
-
-    while(!ifs.eof()) {
-      valBuf.push_back(strtoull((const char*)(&buffer), NULL, 10));
-      std::memset(buffer, 0, sz);
-      ifs.getline(buffer, sz);
-    }
-
-    if(valBuf.size() < 13) {
-      std::cout << "\nERROR: Incomplete AIM counter data in " << path << std::endl;
-      ifs.close();
-      return;
-    }
-
-    aimResults.WriteBytes[index]      = valBuf[0];
-    aimResults.WriteTranx[index]      = valBuf[1];
-    aimResults.ReadBytes[index]       = valBuf[4];
-    aimResults.ReadTranx[index]       = valBuf[5];
-    aimResults.OutStandCnts[index]    = valBuf[8];
-    aimResults.LastWriteAddr[index]   = valBuf[9];
-    aimResults.LastWriteData[index]   = valBuf[10];
-    aimResults.LastReadAddr[index]    = valBuf[11];
-    aimResults.LastReadData[index]    = valBuf[12];
-
-    ifs.close();
+  std::vector<uint64_t> valBuf = xrt_core::device_query<xrt_core::query::aim_counter>(device, dbgIpInfo);
+  aimResults.WriteBytes[index]    = valBuf[0];
```

Comment:
It would be good to have an alternate way of representing the valBuf index instead of using an magic number.